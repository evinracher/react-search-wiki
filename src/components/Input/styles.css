.input-field {
    padding: 0.5rem 1rem;
    border: 1px solid grey;
    border-radius: 0.5rem;
    font-size: 1rem;
}