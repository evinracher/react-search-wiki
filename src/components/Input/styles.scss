.input-field {
    padding: 0.5rem 1rem;
    border: 1px solid grey;
    border-radius: 0.5rem;
    font-size: 1rem;
    width: 20rem;
    box-sizing: border-box;

    &:focus {
        box-shadow: none;
        outline: none;
    }
}