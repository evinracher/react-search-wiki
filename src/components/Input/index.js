import React from 'react';
import './styles.scss';
import PropTypes from 'prop-types';

const Input = ({ placeholder, ...rest }) => {
  return (<input className="input-field" placeholder={placeholder} {...rest}/>)
}

Input.defaultProps = {
  placeholder: 'Input search query',
}

export default Input;