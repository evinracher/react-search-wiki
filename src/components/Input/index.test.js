import React from 'react';
import Input from '.';
import { shallow } from 'enzyme';

const render = (props) => shallow(<Input {...props} />);

describe('Input component', () => {
  let sut;
  let props;

  describe('without props', () => {
    beforeEach(() => {
      sut = render(props);
    });
    
    it('should match snapshot', () => {
      expect(sut).toMatchSnapshot();
    });
  });

  describe('with props', () => {
    beforeEach(() => {
      props = {
        placeholder: 'Input your date',
        name: 'search'
      };
      
      sut = render(props);
    });
    
    it('should match snapshot', () => {
      expect(sut).toMatchSnapshot();
    });
  });


});