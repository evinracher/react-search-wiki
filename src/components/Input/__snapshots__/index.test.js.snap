// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component with props should match snapshot 1`] = `
<input
  className="input-field"
  name="search"
  placeholder="Input your date"
/>
`;

exports[`Input component without props should match snapshot 1`] = `
<input
  className="input-field"
/>
`;
